float32 current
float32 voltage